### PR TITLE
Add option failOnLimitReached

### PR DESCRIPTION
**Pull request by stefanosala**

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -14,31 +16,34 @@ var Request = function(hostname, headers, logLevel) {
   }
 
   this.hostname = hostname;
-  this.headers = headers
+  this.headers = headers;
   this.logger = logger;
   logger.level = logLevel || 0;
+  this.failOnLimitReached = typeof failOnLimitReached === 'undefined' ?
```
Comment (dlebech) [inline lib/request.js]: If `false` is the fallback when the variable is undefined, I think `!!failOnLimitReached` would work here too?

**Comment by dlebech** [comment]
:+1:

**Comment by jmnsf** [comment]
:+1:

**Comment by RickCraig** [comment]
:+1:

**Pull request merged**